src/test/java/io/supertokens/storage/mysql/test/LoggingTest.java

```diff
@@ -308,6 +310,190 @@ public void confirmHikariLoggerClosedOnlyWhenProcessEnds() throws Exception {
         assertFalse(hikariLogger.iteratorForAppenders().hasNext());
     }
 
+    @Test
+    public void testDBPasswordMaskingOnDBConnectionFailUsingConnectionUri() throws Exception {
+        StorageLayer.close();
+        String[] args = { "../" };
+
+        String dbUser = "db_user";
+        String dbPassword = "db_password";
+        String dbName = "db_does_not_exist";
+        String dbConnectionUri = "mysql://" + dbUser + ":" + dbPassword + "@localhost:3306/" + dbName;
+
+        Utils.setValueInConfig("mysql_connection_uri", dbConnectionUri);
+        Utils.commentConfigValue("mysql_user");
+        Utils.commentConfigValue("mysql_password");
+
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+
+        process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.INIT_FAILURE);
+
+        File errorLog = new File(Config.getConfig(process.getProcess()).getErrorLogPath(process.getProcess()));
+
+        boolean dbPasswordMaskedInErrorLog = false;
+
+        try (Scanner errorScanner = new Scanner(errorLog, StandardCharsets.UTF_8)) {
+            while (errorScanner.hasNextLine()) {
+                String line = errorScanner.nextLine();
+                if(line.contains(dbName) && line.contains(dbUser) && !line.contains(dbPassword) && line.contains("********")){
+                    dbPasswordMaskedInErrorLog = true;
+                    break;
+                }
+            }
+        }
+
+        assertTrue(dbPasswordMaskedInErrorLog);
+        process.kill();
+
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
+
+    @Test
+    public void testDBPasswordMaskingOnDBConnectionFailUsingCredentials() throws Exception {
+        StorageLayer.close();
+        String[] args = { "../" };
+
+        String dbUser = "db_user";
+        String dbPassword = "db_password";
+        String dbName = "db_does_not_exist";
+
+        Utils.setValueInConfig("mysql_user", dbUser);
+        Utils.setValueInConfig("mysql_password", dbPassword);
+        Utils.setValueInConfig("mysql_database_name", dbName);
+
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+
+        process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.INIT_FAILURE);
+
+        File errorLog = new File(Config.getConfig(process.getProcess()).getErrorLogPath(process.getProcess()));
```

Comment:
this is not the best way to observe error logs. The better way is to overwrite the logging mechanicm to write to a buffer you create in this test and then just read that. There are tests in the core that do this already like this one: https://github.com/supertokens/supertokens-core/blob/master/src/test/java/io/supertokens/test/LoggingTest.java#L217

Reply:
This comment applies to all new tests you have added that do this.